javascripts/2048.js

```diff
@@ -1,32 +1,406 @@
+// from: http://stackoverflow.com/questions/7837456/comparing-two-arrays-in-javascript
+Array.prototype.equals = function (array) {
+  // if the other array is a falsy value, return
+  if (!array) {
+    return false;
+  }
+
+  // compare lengths - can save a lot of time
+  if (this.length != array.length) {
+    return false;
+  }
+
+  for (var i = 0; i < this.length; i++) {
+    // Check if we have nested arrays
+    if (this[i] instanceof Array && array[i] instanceof Array) {
+      // recurse into the nested arrays
+      if (!this[i].equals(array[i])) {
+        return false;
+      }
+    } else if (this[i] != array[i]) {
```

Comment:
This comparison might not be strict enough in some cases. For instance `2 != "2"` returns `false` whereas `2 !== "2"` returns `true` (the type matters when using `===` or `!==`). In general, it is recommended to use the later to avoid well-hidden bugs.